src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,50 @@
+import { useState } from 'react';
+import styles from './Dropdown.module.css';
+import { useOutsideClick } from '../../hooks/useOutsideClick';
+
+export default function Dropdown({
+  list,
+  value,
+  onSelect,
+  placeholder
+}: {
+  list: string[];
+  value: string;
+  onSelect: (value: string) => void;
+  placeholder: string;
+}) {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const handleToggle = () => setIsOpen((prev) => !prev);
+
+  const ref = useOutsideClick<HTMLDivElement>(() => setIsOpen(false));
+
+  return (
+    <div className={styles.wrapper} ref={ref}>
+      <button
+        type="button"
+        autoFocus={value === ''}
+        className={`${styles.selectBox} ${isOpen && styles.open} ${value ? styles.selected : styles.unSelected}`}
+        onClick={handleToggle}
+      >
+        {value || placeholder}
+      </button>
+      {isOpen && (
+        <ul className={styles.list}>
+          {list.map((company) => (
+            <li
+              key={company}
+              className={styles.item}
+              onClick={() => {
+                onSelect(company);
+                setIsOpen((prev) => !prev);
+              }}
```

Comment:
```suggestion
              onClick={() => {
                onSelect(company);
                setIsOpen(false);
              }}
```
드롭다운 요소를 클릭했을 때는 토글 보다는 드롭다운의 열림 상태를 false하는게 조금 더 직관적인 것 같아요!!

여기 로직도 `handleDropdownItemClick` 같이 함수로 한번 분리해도 좋을 것 같네요🙃